rotkehlchen/exchanges/htx.py

```diff
@@ -188,24 +221,235 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+            result.extend(output)
+            try:
+                query_options['from'] = output[-1]['id'] + 1  # from is inclusive, so we want after
+                created_at = output[-1]['created-at']  # the results are in descending order
+            except KeyError as e:
+                log.error(f'Missing expected keys {e!s} while iterating for HTX deposits and withdrawals')  # noqa: E501
+                break
+
+            if len(output) < PAGINATION_LIMIT or created_at < start_ts:
+                break
+
+        return result
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            query_for: AssetMovementCategory,
+    ) -> list[AssetMovement]:
+        """
+        Process deposits/withdrawals from HTX. If any asset is unknown or we fail to
+        deserialize an entry the error is logged and we skip the entry.
+
+        This logic processes all entries returned by the API, discarding those that do not fall
+        within the range start_ts <= timestamp <= end_ts. Additionally, it does not terminate the
+        querying process until it encounters an created_at value that is lower than the start_ts.
+        """
+        log.debug(f'querying HTX online with {start_ts=}-{end_ts=}')
+        try:
+            raw_data = self._paginated_query(
+                endpoint='/v1/query/deposit-withdraw',
+                options={
+                    'type': 'deposit' if query_for == AssetMovementCategory.DEPOSIT else 'withdraw',  # noqa: E501
+                    'size': PAGINATION_LIMIT,
+                    'direct': 'next',  # sorted by descending order
+                },
+                start_ts=start_ts,
+            )
+        except RemoteError:
+            log.error('Failed to query HTX api for deposits and withdrawals')
+            return []
+
+        movements = []
+        for movement in raw_data:
+            if (timestamp_raw := movement.get('created-at')) is not None:
+                timestamp = ts_ms_to_sec(TimestampMS(int(timestamp_raw)))
+                if not start_ts <= timestamp <= end_ts:
+                    continue  # skip if is not in range, could come from last page
+            else:
+                log.error(f'Could not find timestamp for {movement}. Skipping...')
+                continue
+
+            try:
+                coin = asset_from_htx(movement['currency'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["currency"]}. Skipping...')  # noqa: E501
+                continue
+
+            try:
+                movements.append(AssetMovement(
+                    timestamp=timestamp,
+                    location=Location.HTX,
+                    category=AssetMovementCategory.DEPOSIT if movement['type'] == 'deposit' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
```

Comment:
1. If the function already passed the category what you do here is silly. You already have this.
2. you don't need that either. Is the other `type`, called `withdrawal`? Then you can just do the following
```suggestion
                    category=AssetMovementCategory[movement['type'].upper()],
```